src/mem/remoteallocator.h

```diff
@@ -22,16 +22,34 @@ namespace snmalloc
       std::atomic<Remote*> next{nullptr};
     };
 
-    alloc_id_t allocator_id;
+    /*
+     * We embed the size class in the bottom 8 bits of an allocator ID (i.e.,
+     * the address of an Alloc's remote_alloc's message_queue; in practice we
+     * only need 7 bits, but using 8 is conjectured to be faster).  The hashing
+     * algorithm of the Alloc's RemoteCache already ignores the bottom
+     * "initial_shift" bits, which is, in practice, well above 8.  There's a
+     * static_assert() over there that helps ensure this stays true.
+     *
+     * This does mean that we might have message_queues that always collide in
+     * the hash algorithm, if they're within "initial_shift" of each other. Such
+     * pairings will substantially decrease performance and so we prohibit them
+     * and use SNMALLOC_ASSERT to verify that they do not exist in debug builds.
+     */
+    alloc_id_t alloc_id_and_sizeclass;
```

Comment:
Can this be a class type?  We have some helpers for bitfield extracts, so we could easily provide an abstract interface for getting the two bits.

Reply:
So this is pretty contained in a single class already adding another layer of abstraction probably wouldn't buy us much.